spark/src/test/java/org/apache/iceberg/spark/SparkTestBase.java

```diff
@@ -124,4 +126,8 @@ protected void assertEquals(String context, List<Object[]> expectedRows, List<Ob
   protected static String dbPath(String dbName) {
     return metastore.getDatabasePath(dbName);
   }
+
+  protected BaseActions actionsForTable(Table table) {
+    throw new UnsupportedOperationException("Reimplement actionsForTable with Actions.forTable() if needed");
+  }
```

Comment:
If this is only sometimes needed would it make sense to have this be a separate interface? (totally possible I've been writing too much scala code though and that pattern would be annoying to do here)/

Reply:
I actually stewed on this a while as well. We could get compile time safety here but it would cost us adding another interface that would only be used by test classes. I thought about it for a while and just decided a runtime error is fine for devs :) 

But I would be glad to change it to an interface instead, I think it really just comes down to, do you hate runtime errors more than adding classes. In Scala I would probably just include it as a trait in the BaseTable.java file  so It wouldn't bother me so much.

Reply:
I agree with your assessment, thanks for explaining it :)